src/Microsoft.AspNetCore.Mvc.Razor/Internal/PrecompiledViewsFeatureProvider.cs

```diff
@@ -0,0 +1,57 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+using Microsoft.AspNetCore.Mvc.ApplicationParts;
+
+namespace Microsoft.AspNetCore.Mvc.Razor.Internal
+{
+    public class PrecompiledViewsFeatureProvider : IApplicationFeatureProvider<PrecompiledViewsFeature>
+    {
+        public static readonly string PrecompiledResourcePrefix = "__RazorPrecompiledView__.";
+        private const string DllExtension = ".dll";
+        private const string PdbExtension = ".pdb";
+
+        public void PopulateFeature(IEnumerable<ApplicationPart> parts, PrecompiledViewsFeature feature)
+        {
+            foreach (var assemblyPart in parts.OfType<AssemblyPart>())
+            {
+                AddPrecompiledViews(feature, assemblyPart.Assembly);
+            }
+        }
+
+        private void AddPrecompiledViews(PrecompiledViewsFeature feature, Assembly assembly)
+        {
+            var resourceNames = new HashSet<string>(assembly.GetManifestResourceNames(), StringComparer.Ordinal);
+            foreach (var resourceName in resourceNames)
+            {
+                if (resourceName.StartsWith(PrecompiledResourcePrefix, StringComparison.Ordinal) &&
+                    resourceName.EndsWith(".dll", StringComparison.Ordinal))
+                {
+                    var type = ReadResource(assembly, resourceName);
+                    var viewPath = resourceName.Substring(
```

Comment:
Yup. Probably need to normalize the slashes as part of precompilation.

![image](https://cloud.githubusercontent.com/assets/174281/17234013/aeca0e7a-54e9-11e6-8fe0-0bdac3d311d0.png)